### DISCUSSION TITLE
Zavala 3.0 is now in TestFlight

### DISCUSSION THREAD

**Discussion by vincode-io** [vincode-io/Zavala#254] [General]
[Help test Zavala](https://testflight.apple.com/join/iFbND23m)

On the surface it might not look like this should be a major software release. There are only a couple of new features. A lot has changed under the hood though. I adopted Swift Structured Concurrency. Long story short, this will reduce the number of very difficult to debug, runtime crashes. So Zavala is now more stable than ever. Doing this was a lot of work and anytime that much work is involved, bugs inevitably creep in. I think I've found all the bugs, but developers always think that. ;-)

Fun stuff that was added:

* Apple Intelligence Writing Tools are now supported when running macOS 15.1 or iOS 18.1 (currently Beta only) on devices that support Apple Intelligence. Writing Tools use Artificial Intelligence to proof read or rewrite your sentences. For example, you can tell Writing Tools to make your notes you took at a work meeting more professional and it will replace your text with a more professional version.

* I also added a Group Rows command. Select one or more contiguous rows and use Group Rows to add an empty row and indent the selected rows under it.

* Sorting rows is now also possible in Zavala. Select two or more contiguous rows and select Sort Rows to have them sorted alphabetically. Like most things this is animated and provides a kinda cool effect if you are sorting enough items.

As always testing and feedback are always appreciated. Zavala is an Open Source project and testers are a big part of the process of getting an Open Source project ready to be released. Zavala is free and always will be. If you are a Zavala user and want to give back to the project, providing feedback is the way to do it!

**Comment by ghost** [comment]
Hi Maurice 🙂

I’ve been checking the new version, unfortunately I do not have any serious work for it right now, but for what I can tell, I haven’t found bugs yet… Although I would suggest to work a bit on the animation during selecting a few lines on the iOS / iPadOS version of the app. It’s a bit weird I think. I mean the moment when I select a few lines and long touch to get a context menu.

**Comment by vincode-io** [comment]
@mikezelazny Is the fact that only one of the rows gets highlighted when showing the context menu for multiple items or is it something else?

**Reply by ghost** [reply]
Exactly this. 🙂

**Reply by vincode-io** [reply]
I agree that this could be better. I'm struggling with the design though. I can change what you see for the context menu preview (the single line selected currently). But what would I change it to? You can select non-contiguous rows and I can't show them all if a lot of rows are selected.

Let me know if you have any ideas.

**Reply by ghost** [reply]
What about Reminders’ style?

https://github.com/user-attachments/assets/58d17727-6b11-4242-ade2-90b1e330c331

**Reply by vincode-io** [reply]
Good example. This won't happen in time for 3.0 to be released, but I've created an issue to track it in a future release. https://github.com/vincode-io/Zavala/issues/255

**Reply by ghost** [reply]
Reminders, Notes, Files, they all do the same. I’m glad I helped.
